### PR TITLE
Rendered data on the PO Details page ( #1xhkf57 ).

### PR DESCRIPTION
**Pull request by meet-aniket**

### CODE REVIEW

File: src/router/index.ts [hunk 1]
```diff
@@ -54,15 +54,15 @@ const routes: Array<RouteRecordRaw> = [
     beforeEnter: authGuard
   },
   {
-    path: '/purchase-orders',
-    name: 'PurchaseOrders',
-    component: PurchaseOrders,
+    path: "/purchase-order-detail/:slug",
```
Comment (adityasharma7) [inline src/router/index.ts:57]: Please revert unwanted changes

File: src/components/PurchaseOrderItem.vue [hunk 2]
```diff
@@ -26,8 +26,23 @@ export default defineComponent({
   },
   props: ["purchaseOrder"],
   methods: {
-    async viewPurchaseOrder () {
-      this.router.push({ path: `/purchase-order/${this.purchaseOrder.orderId}` })
+    async getOrderDetail(orderId?: any) {
+      const payload = {
```
Comment (adityasharma7) [inline src/components/PurchaseOrderItem.vue:30]: I think we should decide in action whether to make an API call or not. We should pass orderId, and in action we decide whether we want extract from existing list or make API call

File: src/store/modules/order/actions.ts [hunk 3]
```diff
@@ -33,6 +33,28 @@ const actions: ActionTree<OrderState, RootState> = {
     }
     return resp;
   },
+  async getOrderDetail({ commit, state }, { payload, orderId }) {
+    let resp;
+    const current = state.current as any
+    if (current.length && current[0]?.doclist && current[0]?.doclist.docs[0].orderId === orderId)
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:39]: Improve the schema, current should have orderId and other common attribute while items should have complete list with total

File: src/store/modules/order/getters.ts [hunk 4]
```diff
@@ -8,6 +8,9 @@ const getters: GetterTree<OrderState, RootState> = {
   },
   isScrollable(state) {
     return state.purchaseOrders.list.length > 0 && state.purchaseOrders.list.length < state.purchaseOrders.total
+  },
+  getOrderDetail(state) {
```
Comment (adityasharma7) [inline src/store/modules/order/getters.ts:12]:
```suggestion
  getCurrent(state) {
```

File: src/store/modules/order/mutation-types.ts [hunk 5]
```diff
@@ -1,2 +1,3 @@
 export const SN_ORDER = 'order'
 export const ORDER_PRCHS_ORDRS_UPDATED = SN_ORDER + '/PRCHS_ORDRS_UPDATED'
+export const ORDER_PRCHS_DTAIL_UPDATED = SN_ORDER + '/PRCHS_ORDRS_DTAL_UPDATED'
```
Comment (adityasharma7) [inline src/store/modules/order/mutation-types.ts:3]:
```suggestion
export const ORDER_CURRENT_UPDATED = SN_ORDER + '/PRCHS_CURRENT_UPDATED'
```

File: src/views/PurchaseOrderDetail.vue [hunk 6]
```diff
@@ -132,6 +132,12 @@ export default defineComponent({
     IonTitle,
     IonToolbar,
   },
+  computed: {
+    ...mapGetters({
+      orderDetail: 'order/getOrderDetail',
```
Comment (adityasharma7) [inline src/views/PurchaseOrderDetail.vue:137]:
```suggestion
      order: 'order/getCurrent',
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/actions.ts [hunk 7]
```diff
@@ -33,6 +33,54 @@ const actions: ActionTree<OrderState, RootState> = {
     }
     return resp;
   },
+  async getOrderDetail({ commit, state }, { orderId }) {
+    let resp;
+
+    const current = state.current as any
+    const orders = state.purchaseOrders.list as any
+
+    if (current.length && current[0]?.orderId === orderId) return current;
```
Comment (ymaheshwari1) [inline src/store/modules/order/actions.ts:42]:
```suggestion
    if (current.length && current[0]?.orderId === orderId) { return current }
```

File: src/store/modules/order/actions.ts [hunk 8]
```diff
@@ -33,6 +33,54 @@ const actions: ActionTree<OrderState, RootState> = {
...
+            `orderTypeId: PURCHASE_ORDER AND orderId: ${orderId}`
+          ]
+        }
+      }
+      resp = await OrderService.fetchPODetail(payload);
+
+      if (resp.status === 200 && !hasError(resp) && resp.data.grouped) {
+        const order = resp.data.grouped.orderId.groups[0].doclist.docs
+
+        this.dispatch('product/fetchProductInformation', { order: order });
```
Comment (ymaheshwari1) [inline src/store/modules/order/actions.ts:73]:
```suggestion
        this.dispatch('product/fetchProductInformation', { order });
```

File: src/store/modules/order/mutation-types.ts [hunk 9]
```diff
@@ -1,2 +1,3 @@
 export const SN_ORDER = 'order'
 export const ORDER_PRCHS_ORDRS_UPDATED = SN_ORDER + '/PRCHS_ORDRS_UPDATED'
+export const ORDER_CURRENT_UPDATED = SN_ORDER + '/PRCHS_CURRENT_UPDATED'
```
Comment (ymaheshwari1) [inline src/store/modules/order/mutation-types.ts:3]: mismatch between mutation names.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/OrderState.ts [hunk 10]
```diff
@@ -2,5 +2,6 @@ export default interface OrderState {
   purchaseOrders: {
     list: any;
     total: number;
-  }
+  },
+  current: any
```
Comment (adityasharma7) [inline src/store/modules/order/OrderState.ts:6]:
```suggestion
  current: {
    orderId: string;
    externalOrderId: string;
    items: []
  }
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**